backend/src/dao/CoffeeChatDao.ts

```diff
@@ -0,0 +1,102 @@
+import { v4 as uuidv4 } from 'uuid';
+import { memberCollection, coffeeChatsCollection } from '../firebase';
+import { DBCoffeeChat } from '../types/DataTypes';
+import { getMemberFromDocumentReference } from '../utils/memberUtil';
+import BaseDao from './BaseDao';
+
+async function materializeCoffeeChat(dbCoffeeChat: DBCoffeeChat): Promise<CoffeeChat> {
+  const member1 = await getMemberFromDocumentReference(dbCoffeeChat.members[0]);
+  const member2 = await getMemberFromDocumentReference(dbCoffeeChat.members[1]);
+
+  return {
+    ...dbCoffeeChat,
+    members: [member1, member2]
+  };
+}
+
+async function serializeCoffeeChat(coffeeChat: CoffeeChat): Promise<DBCoffeeChat> {
+  const member1Data = memberCollection.doc(coffeeChat.members[0].email);
+  const member2Data = memberCollection.doc(coffeeChat.members[1].email);
+
+  return {
+    ...coffeeChat,
+    members: [member1Data, member2Data]
+  };
+}
+
+export default class CoffeeChatDao extends BaseDao<CoffeeChat, DBCoffeeChat> {
+  constructor() {
+    super(coffeeChatsCollection, materializeCoffeeChat, serializeCoffeeChat);
+  }
+
+  /**
+   * Creates a new coffee chat for member
+   * @param coffeeChat - Newly created CoffeeChat object.
+   * If provided, the object uuid will be used. If not, a new one will be generated.
+   * The pending field will be set to true by default.
+   */
+  async createCoffeeChat(coffeeChat: CoffeeChat): Promise<CoffeeChat> {
+    const coffeeChatWithUUID = {
+      ...coffeeChat,
+      status: 'pending' as Status,
+      uuid: coffeeChat.uuid ? coffeeChat.uuid : uuidv4()
+    };
+    return this.createDocument(coffeeChatWithUUID.uuid, coffeeChatWithUUID);
+  }
+
+  /**
+   * Gets the coffee chats
+   * @param uuid - DB uuid of coffee chat
+   */
+  async getCoffeeChat(uuid: string): Promise<CoffeeChat | null> {
+    return this.getDocument(uuid);
+  }
+
+  /**
+   * Gets all coffee chats
+   */
+  async getAllCoffeeChats(): Promise<CoffeeChat[]> {
+    return this.getDocuments();
+  }
+
+  /**
+   * Updates a coffee chat
+   * @param coffeeChat - updated Coffee Chat object
+   */
+  async updateCoffeeChat(coffeeChat: CoffeeChat): Promise<CoffeeChat> {
+    return this.updateDocument(coffeeChat.uuid, coffeeChat);
+  }
+
+  /**
+   * Gets all coffee chat for a user
+   * @param user - user whose coffee chats should be fetched
+   */
+  async getCoffeeChatsByUser(user: IdolMember): Promise<CoffeeChat[]> {
+    return this.getDocuments([
+      {
+        field: 'members',
+        comparisonOperator: 'array-contains',
+        value: memberCollection.doc(user.email)
+      }
+    ]);
+  }
+
+  /**
+   * Deletes a coffee chat
+   * @param uuid - DB uuid of CoffeeChat
+   */
+  async deleteCoffeeChat(uuid: string): Promise<void> {
+    await this.deleteDocument(uuid);
+  }
+
+  /**
+   * Deletes all coffee chats for all users
+   */
+  static async clearAllCoffeeChats(): Promise<void> {
```

Comment:
You should be able to use `deleteCollection` here.

https://github.com/cornell-dti/idol/blob/95ee0b88aaaf2b6bc56ee4783b7edf1cf59d84d8/backend/src/utils/firebase-utils.ts#L57-L68